contracts/modules/claims/Incidents.sol

```diff
@@ -0,0 +1,234 @@
+    uint coveredTokenAmount
+  ) external returns (uint claimId, uint payoutAmount) {
+
+    Incident memory incident = incidents[incidentId];
+    address coverOwner;
+    address coveredToken;
+    address underlyingToken;
+    uint sumAssured;
+
+    {
+      QuotationData qd = quotationData();
+      uint coverExpirationDate;
+      uint coverStartDate;
+      bytes4 currency;
+
+      (
+      coveredToken, coverOwner, coverStartDate,
+      coverExpirationDate, sumAssured, currency
+      ) = getCoverDetails(qd, coverId);
+
+      // check ownership & validity
+      require(msg.sender == coverOwner, "Incidents: Not cover owner");
+      require(coverStartDate >= incident.date, "Incidents: Cover start date before the incident");
+      require(coverExpirationDate <= incident.date, "Incidents: Cover end date after the incident");
+
+      {
+        // check covered protocol
+        require(coveredToken == incident.coveredToken, "Incidents: Covered token != incident token");
+
+        // note: technically this check should have happened at cover purchase time
+        underlyingToken = underlyingTokens[coveredToken];
+        address coverCurrencyAddress = claimsReward().getCurrencyAssetAddress(currency);
+        require(coverCurrencyAddress == underlyingToken, "Incidents: Cover asset != underlying asset");
+      }
+
+      // decrese total sum assured
+      qd.subFromTotalSumAssured(currency, sumAssured);
+      qd.subFromTotalSumAssuredSC(coveredToken, currency, sumAssured);
+    }
+
+    {
+      // min/max checks. sumAssured assumes 18 decimals
+      uint coverAmount = sumAssured.mul(1e18);
+      uint maxAmount = coverAmount.div(incident.priceBefore);
+      uint minAmount = maxAmount.mul(20).div(100);
+
+      require(coveredTokenAmount <= maxAmount, "Incidents: Amount exceeds sum assured");
+      require(coveredTokenAmount >= minAmount, "Incidents: Amount is less than 20% of sum assured");
+
+      // coveredTokenAmount / maxAmount * coverAmount
+      payoutAmount = coveredTokenAmount.mul(coverAmount).div(maxAmount);
+    }
+
+    {
+      // mark cover as having a successful claim
+      TokenController tc = tokenController();
+      tc.markCoverClaimOpen(coverId);
+      tc.markCoverClaimClosed(coverId, true);
+
+      // create the claim
+      ClaimsData cd = claimsData();
+      claimId = cd.actualClaimLength();
+      cd.addClaim(claimId, coverId, coverOwner, now);
+      cd.callClaimEvent(coverId, coverOwner, claimId, now);
+      claimPayouts[claimId] = payoutAmount;
+    }
+
+    {
+      Pool p1 = pool();
+      address payable coverOwnerPayable = address(uint160(coverOwner));
+      address payable payoutAddress = memberRoles().getClaimPayoutAddress(coverOwnerPayable);
+
+      // pull depeged tokens and send the payoutAmount
+      IERC20(coveredToken).safeTransferFrom(msg.sender, address(this), coveredTokenAmount);
+      bool success = p1.sendClaimPayout(underlyingToken, payoutAddress, payoutAmount);
```

Comment:
something to keep in mind: this will trigger updateMCR as well in this call

Reply:
ah this calls straight into p1 my bad.

this *will need to trigger updateMCR if we are to keep it consistent to the other payout flow.

gas considerations here